loom/graphics/gfxStateManager.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * ===========================================================================
+ * Loom SDK
+ * Copyright 2011, 2012, 2013
+ * The Game Engine Company, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================
+ */
+
+#include "gfxStateManager.h"
+
+bool gGLStates[GFX_OPENGL_STATE_MAX] = {true, true};
+
+bool Graphics_IsGLStateValid(enum Graphics_GLState state)
+{
+    return gGLStates[(int)state];
+}
+
+void Graphics_InvalidateGLState(enum Graphics_GLState state)
+{
+    gGLStates[(int)state] = false;
+}
+
+void Graphics_SetCurrentGLState(enum Graphics_GLState state)
+{
+    int i;
+    for (i = 0; i < (int)GFX_OPENGL_STATE_MAX; i++)
```

Comment:
Yes, only one can be true at once, but they both can be false (everything uninitialized). And this way is more extensible.